src/script/miniscript.h

```diff
@@ -779,8 +779,11 @@ struct Node {
     //! Check whether there is no satisfaction path that contains both timelocks and heightlocks
     bool CheckTimeLocksMix() const { return GetType() << "k"_mst; }
 
-    //! Do all sanity checks.
-    bool IsSane() const { return IsValid() && IsNonMalleable() && CheckTimeLocksMix() && CheckOpsLimit() && CheckStackSize(); }
+    //! Whether successful non-malleable satisfactions are guaranteed to be valid.
```

Comment:
I think you mean all satisfactions here? Not only the non-malleable ones?

Reply:
Only the non-malleable ones, because CheckOpsLimit and CheckStackSize only take the logic of the non-malleable satisfaction algorithm into account.